### ISSUE TITLE
Import not supported in Brave browser

### ISSUE THREAD

**Issue by Galeforce1** [GarinZ/link-map#21]
I am unable to import Tabs Outliner or Link Map data in Brave browser. The error message says that my browser does not support this feature. But I've found a way of getting my links from Brave back to Chrome (using TO) and then importing them into Link Map and copying them over to Link Map on Brave. Clumsy but it works.

Why is this feature unsupported? Will it be supported in the future? If so, when?

Otherwise it looks good, very similar to TO but with more features generally. It needs more instructions to explain the current feature set.

**Comment by GarinZ** [comment]
Don't worry, I'd like to check the compatibility in Brave browser these days. When I have checked and made it work in Brave, I will infor you in this issue.
About instruction, yes, you are right, it's not enough for users to have a overview of full features set in Link Map, I will improve it.

**Comment by GarinZ** [comment]
@Galeforce1 Oops, I have found the problem and fix it. If you are interested, you could checkout this [commit](https://github.com/GarinZ/link-map/commit/1a3b8e8745e9e54303b9d0940320a22569118069).
It need some time to test for Brave and make the patched version online (need to be reviewed by the audit team from browser ecosystem).
When the new version has been released.,I will update the progress in this issue.

**Comment by Galeforce1** [comment]
Hi Garin, thanks for your reply. I've loaded the 1.0.9 fix, but the import still doesn't work. I get the same message 'Your browser does not support this feature'. This message happens when I try both import functions (TO and Link Map data).

![image](https://user-images.githubusercontent.com/18387332/230751631-3c408bef-17f3-437f-9981-79d752850d89.png)

**Comment by GarinZ** [comment]
Hi @Galeforce1, I have double checked the v1.0.9 in my Brave just now, and it works for me. I checked it in MacOS. Could your please tell me which operating system do you currently use?

And I want to give a remind that when you load v1.0.9, there will two version of Link Map in the Browser(I have uploaded a picture below). Would you please check if you clicked the extension icon point to v1.0.9?

![image](https://user-images.githubusercontent.com/7566103/230758101-94e34246-4b8f-4cd5-a366-b00d5115a5a8.png)

**Comment by Galeforce1** [comment]
You are quite right - I didn't realise there were 2 versions in the browser. Yes, it works fine. I'm using Window 10. Thank you for such a rapid fix!

1. I noticed in the Chrome store that the images show LM with Note Nodes, but I don't seem to have them. Are they the tags?
2. Are you planning to have dividers and containers like it TO? It's just a visual help for organising links?

**Comment by GarinZ** [comment]
1. Yes, tag node is note node. I used to define it as note node.  I should change the description in the store page.
2. I will manage to add this feature. when I use TO, divider is helpful for me too. But in current architecture, It's not that easy. I need sometime to find a way to achieve this.

**Comment by Galeforce1** [comment]
Hi Garin,
This morning I found that LM had 'lost' the TO import I did yesterday. Not sure how this happened. I left Brave open all night and didn't close it at all. It also lost the history of windows/tabs I had open yesterday. It's as if it just didn't save it all. I'm running both TO and LM together - would that cause any conflicts?

So I re-did the TO import and then closed and re-opened Brave, and it was ok, it had saved the TO import and the window/tab history. If it happens again, I will let you know. Meanwhile, I've made a backup by exporting it.

I'm using the tag/notes as dividers - temporarily (see image). When used as a note, it would be good if the text in the tag/note could wrap around so you could read it all. 
![image](https://user-images.githubusercontent.com/18387332/230813771-513f19dc-b670-424b-807b-f48bdbae2ec3.png)

Many thanks.

**Comment by GarinZ** [comment]
Sorry for your lost @Galeforce1 , losing data is the last thing I want to see, And I will take this case seriously. Could  your please tell me details about the problems to help me reproduce this case?
- Did Link Map keep opening until you find data has lost? Or It has been closed and when re-active it this morning, data just dispeared.
- Could your please check the version of Link Map again to make sure the version with no saving data and the version you saved data into is the same version.(Although they are both Link Map, but it is two extension for browser with diffrent ID, and isolation storage space)
- Have you re-installed Link Map? In this case, data will clear by browser implicitly

It's totally ok to run TO and LM  at same time, there wouldn't be any conflict.

---

Wrap the note when it's too long : It's reasonable, I will find a way to optimize it, but achieveing this is not that easy like it seems to be.

**Comment by Galeforce1** [comment]
> Did Link Map keep opening until you find data has lost? Or It has been closed and when re-active it this morning, data just 
dispeared.

No, I think I closed it last night but left Brave open (I always leave Brave open at night), and then re-opened it (LM) this morning. That's when I noticed the import & history had gone.

> Could your please check the version of Link Map again to make sure the version with no saving data and the version you saved data into is the same version.(Although they are both Link Map, but it is two extension for browser with diffrent ID, and isolation storage space)

Good point - I didn't think of that. Now I have only version 1.0.9 installed. I uninstalled the earlier version yesterday after you told me and I realised that there were 2 installed. _I'm pretty sure that the TO import (yesterday) was into version 1.0.9, not the older version, done **after** I had uninstalled the older version._

> Have you re-installed Link Map? In this case, data will clear by browser implicitly

No, I haven't re-installed version 1.0.9 since installing it the 1st time yesterday.

Let's wait and see if it happens again. I don't think there is much I can do to recreate the problem, apart from use LM normally.

**Comment by GarinZ** [comment]
Thanks for providing me these information.
If you encountered the same problem(I hope not), please tell me.
I'm researching if Brave has some special hevaior for extension, and trying to figure out the cause of this case.
